devito/finite_differences/finite_difference.py

```diff
@@ -62,8 +65,25 @@ def wrapper(expr, *args, **kwargs):
     return wrapper
 
 
+def check_symbolic(func):
+    @wraps(func)
+    def wrapper(expr, *args, **kwargs):
+        functions = retrieve_functions(expr)
+        functions = filter_ordered(functions, key=lambda i: i.name)
+        functions = filter_sparse(functions)
+        symbolic_coefficients = any(f.coefficients is 'symbolic' for f in functions)
+        if symbolic_coefficients:
+            expr_dict = expr.as_coefficients_dict()
+            if any(len(expr_dict.keys()) > 1 for item in expr_dict):
+                raise NotImplementedError
+        kwargs['symbolic_coefficients'] = symbolic_coefficients
+        return func(expr, *args, **kwargs)
+    return wrapper
+
+
 @check_input
-def first_derivative(expr, dim, fd_order=None, side=centered, matvec=direct):
+@check_symbolic
+def first_derivative(expr, dim, fd_order=None, side=centered, matvec=direct, **kwargs):
```

Comment:
probably better to have an explicit ``symbolic=False`` kwarg, instead of ``**kwargs``, and make sure that ``check_symbolic`` always adds in the information (be it True or False)

Reply:
Ta, changed.